src/days-2-4-adv/adv.py

```diff
@@ -49,3 +61,86 @@
 # Print an error message if the movement isn't allowed.
 #
 # If the user enters "q", quit the game.
+
+
+def DirectionEval(direction):
+    if hasattr(currentPlayer.room, directionFunctions[direction]):
+        currentPlayer.room = getattr(
+            currentPlayer.room, directionFunctions[direction])
+        return False
+    else:
+        print("Invalid Direction")
+        return True
+
+
+def ItemEval(item, itemsList):
+    return next((y for y in itemsList if y.name.lower() == item), None)
+
+
+def TextEval(text):
+    print(text)
+    if text == "q" or text == "quit":
+        raise AdventureDone
+    elif text in directionFunctions.keys():
+        return DirectionEval(text)
+    elif "score" in text:
+        print(currentPlayer.score)
+        return True
+    elif "grab" in text or "take" in text:
+        if not currentPlayer.room.is_light or not currentPlayer.hasLight:
+            print("Good luck finding that in the dark!")
+            return True
```

Comment:
I had a little problem here in the beginning. There is a lamp in the cave entrance, but I can't pick it up because it's dark and I don't already have a light. I cannot pick up the cake or head north, either, because I have no light and cannot pick anything up. This should be an easy fix, though. Maybe make the outside light or something, and then the player can pick up the lamp and use it to continue onward.